### PR TITLE
Add tabry-generate-fish-complete

### PR DESCRIPTION
**Pull request by pfgray**
This commit adds on-the-fly fish completions, to match the bash generation functionality

This will enable tabry apps to generate fish completions with a command like:

```
<cmd> completion fish
```

which can then be sourced.

This commit also adds that functionality automatically to the `withTabry` nix helper.

### CODE REVIEW

File: sh/fish/tabry_fish.fish [hunk 1]
```diff
@@ -10,15 +10,21 @@ function tabry_completion_init
   complete -c "$cmd" -n "__fish_tabry_check_args_and_file" -a "(__fish_tabry_completions)"
 end
 
+# _tabry_#{uniq_fn_id}_completions() {
+#   TABRY_IMPORTS_PATH=#{esc import_path} _tabry_#{uniq_fn_id}_completions_internal #{esc tabry_bash_executable} #{tabry_bash_arg && esc(tabry_bash_arg)}
+# }
+
```
Comment (evanbattaglia) [inline sh/fish/tabry_fish.fish]: Can/should this be removed or did you intend it for documentation purposes?

File: lib/tabry/shells/fish.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        script = <<~HEREDOC
+          #{script}
+          tabry_completion_init_#{uniq_fn_id} #{esc cmd_name}
+        HEREDOC
+
+        <<~END_FISH_CODE_TEMPLATE + script
+          # The following Autocomplete is for a Tabry-powered command. It was
+          # generated by the command itself. See the documentation located in
+          # #{esc path_to_tabry}/sh/fish/README.md
+        END_FISH_CODE_TEMPLATE
```
Comment (evanbattaglia) [inline lib/tabry/shells/fish.rb:78]: any reason not to merge the two previous heredocs?

**Review by evanbattaglia — COMMENTED** [review]
Thanks! A couple questions

File: sh/fish/tabry_fish.fish (hunk 1 above)
Comment (pfgray) [reply sh/fish/tabry_fish.fish]: Ah, I forgot I had that in there to copy/compare, just removed it

File: lib/tabry/shells/fish.rb (hunk 2 above)
Comment (pfgray) [reply lib/tabry/shells/fish.rb:78]: Still learning how heredoc works, and yeah, I just merged them

**Review by evanbattaglia — APPROVED** [review]
Let me know if you want me to merge this now -- I'm so used to how Gerrit works, I'm confused about how github works with multiple PRs -- the other PR seems to build on this one I think

**Comment by pfgray** [comment]
This one's fine to go in now, so feel free to merge it... https://github.com/evanbattaglia/tabry/pull/12/commits/3826757c77d337f621b3e678ec4fce70f688b749 is based on this one, mostly just to avoid merge conflicts. If we merge this then that PR will only show 1 commit, and will be easier to see the diff (without having to look at a specific commit)

**Pull request merged**